optuna/integration/mlflow.py

```diff
@@ -102,56 +141,122 @@ def __init__(
 
     def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
 
-        # This sets the tracking_uri for MLflow.
-        if self._tracking_uri is not None:
-            mlflow.set_tracking_uri(self._tracking_uri)
-
-        # This sets the experiment of MLflow.
-        mlflow.set_experiment(study.study_name)
+        self._initialize_experiment(study)
 
-        with mlflow.start_run(run_name=str(trial.number), nested=self._nest_trials):
+        with mlflow.start_run(
+            run_id=trial.system_attrs.get(RUN_ID_ATTRIBUTE_KEY),
+            run_name=str(trial.number),
+            nested=self._nest_trials,
+        ):
 
             # This sets the metric for MLflow.
-            trial_value = trial.value if trial.value is not None else float("nan")
-            mlflow.log_metric(self._metric_name, trial_value)
+            self._log_metric(trial.value)
 
             # This sets the params for MLflow.
-            mlflow.log_params(trial.params)
+            self._log_params(trial)
```

Comment:
It is more consistent to have a dictionary instead of the instance of the Trial.
```suggestion
            self._log_params(trial.params)
```